config.js

```diff
@@ -47,11 +47,20 @@ Config.prototype.get = function (name, type, cb, options) {
     }
 
     return results;
-};
+}
+
+Config.prototype.getInt = function (filename, default_value) {
+
+    const full_path = path.resolve(this.root_path, filename);
+    const r = parseInt(cfreader.read_config(full_path, 'value', null, null), 10);
+
+    if (!isNaN(r)) return r;
+    return parseInt(default_value, 10);
```

Comment:
What about possibility to set as default "null" if value is not present?

Reply:
Why null instead of NaN?

The results of the function should be as easy to deal with as possible. Per JS conventions, if you pass in something that can't be coerced into a number, you get a NaN. Anything else can be counted on to be an integer. If we're going to return something else, we should have a very good reason.

Reply:
I can see the argument for null: If you have no config set, you can know it wasn't set. Of course you should handle that through defaults via the param, but often we don't/won't, because we're often dumb, so instead people will do: `config.getInt('thing') || some_default`.

It's hard to argue for allowing people to be dumb though. But the NaN issue is painful if people expect a non-set config to be falsy.

Of course I just typed all that, and then decided to test it, and now I found out I'm wrong:

```
$ node
> parseInt('fool');
NaN
> parseInt('fool') || 10;
10
```
So basically ignore my entire comment and the stuff I wrote on the other ticket that inspired this also.